src/hu/elte/txtuml/stdlib/util/UUID.java

```diff
@@ -0,0 +1,180 @@
+public interface UUID extends ExternalClass {
+
+	/**
+	 * 
+	 * The clock sequence value associated with this UUID.</br>
+	 * The 14 bit clock sequence value is constructed from the clock sequence
+	 * field of this UUID. The clock sequence field is used to guarantee
+	 * temporal uniqueness in a time-based UUID. The clockSequence value is only
+	 * meaningful in a time-based UUID, which has version type 1. If this UUID
+	 * is not a time-based UUID then this method throws
+	 * UnsupportedOperationException.
+	 *
+	 * @return The clock sequence (int) of this UUID
+	 * 
+	 * @throws UnsupportedOperationException
+	 *             - If this UUID is not a version 1 UUID
+	 */
+	public int clockSequence();
+
+	/**
+	 * Compares this UUID with the specified UUID. The first of two UUIDs is
+	 * greater than the second if the most significant field in which the UUIDs
+	 * differ is greater for the first UUID.
+	 *
+	 * @Specified by: compareTo in interface Comparable<UUID>
+	 * 
+	 * @param val
+	 *            - UUID to which this UUID is to be compared
+	 * 
+	 * @return -1, 0 or 1 as this UUID is less than, equal to, or greater than
+	 *         val (int)
+	 */
+	public int compareTo(UUID val);
+
+	/**
+	 * Compares this object to the specified object. The result is true if and
+	 * only if the argument is not null, is a UUID object, has the same variant,
+	 * and contains the same value, bit for bit, as this UUID.
+	 *
+	 * @param obj
+	 *            - The object to be compared
+	 * 
+	 * @return true if the objects are the same; false otherwise
+	 * 
+	 */
+	public boolean equals(Object obj);
+
+	/**
+	 * Creates a UUID from the string standard representation as described in
+	 * the toString() method.
+	 *
+	 * @param str
+	 *            - A string that specifies a UUID
+	 * 
+	 * @return A UUID with the specified value
+	 * 
+	 * @throws IllegalArgumentException
+	 *             - If name does not conform to the string representation as
+	 *             described in toString()
+	 */
+	public static UUID fromString(String str) {
+		return new UUIDImplementation(java.util.UUID.fromString(str));
+	}
+
+	/**
+	 * Returns the least significant 64 bits of this UUID's 128 bit value.
+	 *
+	 * @return The least significant 64 bits of this UUID's 128 bit value
+	 */
+	public long getLeastSignificantBits();
+
+	/**
+	 * Returns the most significant 64 bits of this UUID's 128 bit value.
+	 *
+	 * @return The most significant 64 bits of this UUID's 128 bit value
+	 */
+	public long getMostSignificantBits();
+
+	/**
+	 * Returns a hash code for this UUID.
+	 *
+	 * @return A hash code value for this UUID
+	 */
+	public int hashCode();
+
+	/**
+	 * Static factory to retrieve a type 3 (name based) UUID based on the
+	 * specified byte array..
+	 *
+	 * @param name
+	 *            - A byte array to be used to construct a UUID
+	 * 
+	 * @return A UUID generated from the specified array
+	 */
+	/*
+	 * public static UUID nameUUIDFromBytes(Collection<Byte> name) {
```

Comment:
we agreed that we will not implement that static method, I just commented it out instead of removing it from the interface...